### PR TITLE
Add Car controller tests and Car service tests

### PR DESCRIPTION
**Pull request by Recdt**

### CODE REVIEW

File: src/test/java/online/carsharing/service/CarServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        carResponseDto.setModel(CAR_MODEL);
+        carResponseDto.setBrand(CAR_BRAND);
+        carResponseDto.setType(Type.SEDAN);
+        carResponseDto.setDailyFee(DEFAULT_DAILY_FEE);
+
+        when(carMapper.toDto(any())).thenReturn(carResponseDto);
+
+        List<CarResponseDto> result = carService.findAll(Pageable.unpaged());
+        assertNotNull(result);
+        assertEquals(result.size(), 1);
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/CarServiceImplTest.java]: magic number

File: src/test/java/online/carsharing/service/CarServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        assertEquals(result.get(0).getModel(), carResponseDto.getModel());
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/CarServiceImplTest.java]: magic number

Comment (nklimovych) [reply src/test/java/online/carsharing/service/CarServiceImplTest.java]: use getFirst()

File: src/test/java/online/carsharing/service/CarServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        assertEquals(result.getModel(), carResponseDto.getModel());
+    }
+
+    @Test
+    @DisplayName("Verify deleteById() method works")
+    void deleteById_ValidId_Success() {
+        when(carRepository.existsById(DEFAULT_ID)).thenReturn(true);
+        doNothing().when(carRepository).deleteById(DEFAULT_ID);
+        carService.deleteById(DEFAULT_ID);
+        verify(carRepository, times(1)).deleteById(DEFAULT_ID);
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/CarServiceImplTest.java]: same

File: src/test/java/online/carsharing/service/CarServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        car.setBrand(CAR_BRAND);
+        car.setType(Type.SEDAN);
+        car.setInventory(DEFAULT_INVENTORY);
+        car.setDailyFee(DEFAULT_DAILY_FEE);
+
+        CarResponseDto carResponseDto = new CarResponseDto();
+        carResponseDto.setModel(CAR_MODEL);
+        carResponseDto.setBrand(CAR_BRAND);
+        carResponseDto.setType(Type.SEDAN);
+        carResponseDto.setDailyFee(DEFAULT_DAILY_FEE);
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/CarServiceImplTest.java]: It would be better to add these entities to the setUp() method for reusability

**Review by nklimovych — DISMISSED** [review]
Good Job!

**Review by azzilian — DISMISSED** [review]

**Review by nklimovych — APPROVED** [review]

**Review by HAIOVYI — APPROVED** [review]
GJ!

**Pull request merged**